snp_blending/urdf/workcell.xacro

```diff
@@ -0,0 +1,246 @@
+  <xacro:property name="control_boxes_width" value="${38*0.0254}"/>
+  <xacro:property name="control_boxes_length" value="${130*0.0254}"/>
+  <xacro:property name="table_height" value="${33.875*0.0254}"/>
+  <xacro:property name="table_width" value="${48*0.0254}"/>
+  <xacro:property name="table_length" value="${30*0.0254}"/>
+
+  <link name="floor"/>
+
+  <link name="platform">
+    <visual>
+      <geometry>
+        <box size="${96*0.0254} ${72*0.0254} ${platform_height}"/>
+      </geometry>
+      <material name="dark_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${96*0.0254} ${72*0.0254} ${platform_height}"/>
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="robot_mount">
+    <visual>
+      <geometry>
+        <box size="${robot_mount_width} ${robot_mount_width} ${robot_mount_height}"/>
+      </geometry>
+      <material name="dark_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${robot_mount_width} ${robot_mount_width} ${robot_mount_height}"/>
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="pushcorp_box_support_1">
+    <visual>
+      <geometry>
+        <box size="${1.5*0.0254} ${1.5*0.0254} ${40.5*0.0254}" />
+      </geometry>
+      <material name="dark_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${1.5*0.0254} ${1.5*0.0254} ${40.5*0.0254}" />
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="pushcorp_box_support_2">
+    <visual>
+      <geometry>
+        <box size="${1.5*0.0254} ${1.5*0.0254} ${40.5*0.0254}" />
+      </geometry>
+      <material name="dark_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${1.5*0.0254} ${1.5*0.0254} ${40.5*0.0254}" />
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="pushcorp_box">
+    <visual>
+      <geometry>
+        <box size="${pushcorp_box_length} ${pushcorp_box_width} ${pushcorp_box_height}"/>
+      </geometry>
+      <material name="light_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${pushcorp_box_length} ${pushcorp_box_width} ${pushcorp_box_height}"/>
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="control_boxes">
+   <visual>
+      <geometry>
+        <box size="${control_boxes_length} ${control_boxes_width} ${control_boxes_height}"/>
+      </geometry>
+      <material name="light_gray"/>
+    </visual>
+    <collision>
+      <geometry>
+        <box size="${control_boxes_length} ${control_boxes_width} ${control_boxes_height}"/>
+      </geometry>
+    </collision>
+  </link>
+
+  <link name="buffy_tcp"/>
+  <joint name="buffy_tcp_joint" type="fixed">
+    <parent link="tool0"/>
+    <child link="buffy_tcp"/>
+    <origin xyz="0.292 0 0.19" rpy="0 ${pi/2} 0"/>
+  </joint>
+
+  <joint name="floor_to_platform" type="fixed">
+    <parent link="floor"/>
+    <child link="platform"/>
+    <origin xyz="0 0 ${platform_height/2}" rpy="0 0 0"/>
+  </joint>
```

Comment:
We probably want to define a .yaml file to calibrate the location of the work table (platform link?) relative to the robot base frame. We would probably use the robot controller itself to set up this frame, then just copy the values to this file.